### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by KatIsCoding**
In this update, forms for creating new posts, new comments to those posts and liking posts have been added

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -11,4 +11,18 @@ def show
     @user = User.find(@user_id)
     @post = Post.find(@post_id)
   end
+
```
Comment (oloomoses) [inline app/controllers/posts_controller.rb:14]: - [ ] Kindly add a `new` method for instantiating a new post.

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -11,4 +11,18 @@ def show
...
+  def create
+    Post.create(author: UsersController.current_user, title: params[:title], text: params[:text])
+  end
```
Comment (oloomoses) [inline app/controllers/posts_controller.rb]: - [ ] To create a new post, it is a requirement that you use strong params. Kindly read about strong params [here](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides)
- [ ] Kindly use the current user to create a new post. the code should look like below.
```ruby
   def create
      @post = @current_user.posts.new(post_params)

      respond_to do |format|
        if @post.save
          format.html { redirect_to @post }
        else
          format.html { render :new }
        end
      end
    end
```

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -11,4 +11,18 @@ def show
...
+
+  def create_comment
+    current_post = Post.find(params[:id])
+    current_post.comments.create(author: UsersController.current_user, text: params[:text])
+  end
```
Comment (oloomoses) [inline app/controllers/posts_controller.rb]: - [ ] Creating a new comment should be done in the `comments_controller`. Kindly create one and add `new` and `create` methods. The methods should look like below:
```ruby
                              def create
    @comment = @current_user.comments.new(params)
                      @comment.post_id = params[:post_id]

                                    if @comment.save
      redirect_to                                           post_                          path(@comment.post)
    else
      @article = Post.find(params[:post_id])
                                                                                           redirect_to post_path(@post)
    end
  end
```

File: app/controllers/posts_controller.rb [hunk 4]
```diff
@@ -11,4 +11,18 @@ def show
...
+
+  def add_like
+    current_post = Post.find(params[:id])
+    Like.create(author: UsersController.current_user, post: current_post)
+  end
```
Comment (oloomoses) [inline app/controllers/posts_controller.rb]: - [ ] Creating a new like should be done in the `likes_controller`. Kindly create one and add `new` and `create` methods. The methods should look like below:
```ruby
                              def create
    @like = @current_user.comments.new(params)
                      @comment.post_id = params[:post_id]

                                    if @comment.save
      redirect_to                                           post_                          path(@comment.post)
    else
      @article = Post.find(params[:post_id])
                                                                                           redirect_to post_path(@post)
    end
  end
```

File: app/views/posts/show.html.erb [hunk 5]
```diff
@@ -1,6 +1,7 @@
 <main>
 <div class="posts-container">
-<%= render :partial => "shared/post_container", locals: { :post => @post }%>
+<%= render :partial => "shared/post_container", locals: { :post => @post, onClick: "" }%>
 <%= render :partial => "shared/comments_container", locals: { :comments => @post.comments, :limit=> 10 }%>
+<%= render :partial => "shared/comment_form"%>
 </div>
```
Comment (oloomoses) [inline app/views/posts/show.html.erb:6]: - [ ] The new post form should be in `views/posts/new.html.erb`.
- [ ] Kindly ensure that a user can create a post, add comments, and a like.

File: config/routes.rb [hunk 6]
```diff
@@ -1,7 +1,9 @@
 Rails.application.routes.draw do
   resources :users, only: [:index, :show] do
-    resources :posts, only: [:index, :show]
+    resources :posts, only: [:index, :show, :create, :comment]
   end
+  post '/users/:user_id/posts/:id', to: 'posts#create_comment'
+  post '/like/:id', to: 'posts#add_like'
```
Comment (oloomoses) [inline config/routes.rb]: - [ ] Kindly update the routes for creating likes and comments accordingly

**Review by oloomoses — CHANGES_REQUESTED** [review]
Hi @KatIsCoding, 


Great job on the linter configuration and the readme file. 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

_Slack: Oloo Moses_
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by arslanbisharat — APPROVED** [review]
# Approved 👏 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

To Highlight: 
- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of code reviews per project which you can see from your dashboard. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

**Pull request merged**